src/index.ts

```diff
@@ -476,33 +576,37 @@ export class UnleashClient extends TinyEmitter {
                     this.emit(EVENTS.RECOVERED);
                 }
 
-                if (response.ok && response.status !== 304) {
+                if (!response.ok) {
+                    if (response.status === 304) {
+                        this.storeLastRefreshTimestamp();
+                    } else {
+                        console.error(
+                            'Unleash: Fetching feature toggles did not have an ok response'
+                        );
+                        this.sdkState = 'error';
+                        this.emit(EVENTS.ERROR, {
+                            type: 'HttpError',
+                            code: response.status,
+                        });
+
+                        this.lastError = {
+                            type: 'HttpError',
+                            code: response.status,
+                        };
+                    }
+                } else {
```

Comment:
Success-first statement is easier to read here, because there is no early return.

```
if (response.ok) {
} else if (response.status === 304) {
} else {}
```

(Using the fact that [`response.ok` is a shorthand](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok))